pkg/virt-controller/watch/vmi.go

```diff
@@ -589,13 +617,23 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 		}
 		c.recorder.Eventf(vmi, k8sv1.EventTypeNormal, SuccessfulCreatePodReason, "Created virtual machine pod %s", pod.Name)
 		return nil
+	} else {
+		if vmi.IsProvisioning() && !wffc {
+			err := c.clientset.CoreV1().Pods(vmi.Namespace).Delete(pod.Name, &v1.DeleteOptions{})
```

Comment:
it might be wise for us to put a special label or annotation on these temp pods so we can differentiate them and filter them differently from the vmi pods.

Reply:
Added annotation